tests/lib/modem_trace/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(modem_trace_test)
+
+# create mock
+cmock_handle(${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nrfx/drivers/include/nrfx_uarte.h )
+cmock_handle(${ZEPHYR_SEGGER_MODULE_DIR}/SEGGER/SEGGER_RTT.h )
+
+# generate runner for the test
+test_runner_generate(src/modem_trace_test.c)
+
+target_include_directories(app PRIVATE src)
+
+# add test file
+target_sources(app PRIVATE src/modem_trace_test.c)
+
+# add unit under test
+target_sources(app PRIVATE ${NRF_DIR}/lib/nrf_modem_lib/modem_trace.c)
+
+# include paths
+target_include_directories(app PRIVATE ${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nrfx)
+target_include_directories(app PRIVATE ${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nrfx/mdk)
+target_include_directories(app PRIVATE ${ZEPHYR_CMSIS_MODULE_DIR}/CMSIS/Core/Include/)
+target_include_directories(app PRIVATE ${ZEPHYR_SEGGER_MODULE_DIR}/Config/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/modules/hal_nordic/nrfx)
+target_include_directories(app PRIVATE ${NRF_DIR}/include/modem/)
+target_include_directories(app PRIVATE ${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/ )
+
+# Define the device. Otherwise an "unknown device" warning will be triggered when compiling
+# nrfx_config.h
+target_compile_definitions(app PRIVATE NRF9160_XXAA)
+# Define NRFX_DECLARE_ONLY to simplify generation of mocks. If this is not done, then the
+# the compiler would be unable to compile the generated mocks for nrfx uart driver.
+target_compile_definitions(app PRIVATE NRFX_DECLARE_ONLY)
+# Since Device Tree is not available when running unit tests, we need to provide the hardware
+# details that otherwise would be fetched by the UUT from the Device Tree.
+target_compile_definitions(app PRIVATE DT_N_S_uart_1_P_tx_pin=17)
+target_compile_definitions(app PRIVATE DT_N_S_uart_1_P_rx_pin=18)
+
+if(BOARD STREQUAL qemu_cortex_m3)
+    target_compile_definitions(app PRIVATE DT_N_S_soc_S_uart_4000d000=DT_N_S_uart_1)
+endif()
+
+target_compile_definitions(app PRIVATE NRFX_UARTE1_ENABLED)
+# Define the Up and Down buffer count needed for compiling the Segger RTT headers.
+target_compile_definitions(app PRIVATE
+        CONFIG_SEGGER_RTT_MAX_NUM_UP_BUFFERS=3
+        CONFIG_SEGGER_RTT_MAX_NUM_DOWN_BUFFERS=3)
```

Comment:
tests have `extra_configs` fields for this purpose.

Reply:
oh very nice. I was not aware of that. Will get rid of these then.

Reply:
If they are hardcoded, why aren't they added to `prj.conf` then?

Reply:
If I do that, I get an error that says I am attempting to set a KConfig option which is not defined.

Reply:
> Compilations fail when using west and twister. 

Not sure I understand the `and` here ?
Are you trying to invoke `twister` using `west`, cause that's not possible afaik.

> And apparently the extra_configs no longer gets parsed by the west's build script?

It never has. This is purely for `twister`.

> Hence we have to depend on twister (if I get this to work) to run this test. 

Correct, which is why it's wise to use `twister` for tests, and if not, pass those settings manually.
Just like in this `uart` test case: https://github.com/nrfconnect/sdk-zephyr/blob/main/tests/drivers/uart/uart_pm/testcase.yaml
where all tests needs adjustment to some Kconfig settings.

If running the uart test using `west` one must manually specify those flags so it corresponds to the specific test case you want to execute (or use `twister` itself).

Reply:
> Are you trying to invoke twister using west, cause that's not possible afaik.

No. What I meant was, I run unit tests in two ways. 
a) `west build -t run`
b) `twister -T .`

>It never has. This is purely for twister.
There certainly has been some work done in west's build script to parse the `extra_configs` parameter. But it seems partially done. See https://github.com/zephyrproject-rtos/zephyr/blob/e8df8a579033be238beb774e4909d637da649f3a/scripts/west_commands/build.py#L249

Reply:
@tejlmand  Moved the KConfig options from CMakelists.txt to testcase.yaml and KConfig as discussed